### PR TITLE
Refactored process of creating entity

### PR DESCRIPTION
**Pull request by hAdamm**

### CODE REVIEW

File: src/EntityManager.php [hunk 1]
```diff
@@ -77,13 +77,28 @@ public static function isEntityBasedInRepository(\obo\Entity $entity) {
...
         $entity->setDataStorage(self::dataStorage());
         return $entity;
     }
 
+    /**
+     * @internal
+     * @param mixed $primaryPropertyValue
+     * @return \obo\Entity
+     */
+    protected static function mapedEntity($primaryPropertyValue) {
```
Comment (lordjim666) [inline src/EntityManager.php]: mapped

File: src/EntityManager.php [hunk 2]
```diff
@@ -95,17 +110,15 @@ public static function emptyEntity() {
...
      */
     public static function entityWithPrimaryPropertyValue($primaryPropertyValue, $ignoreSoftDelete = false) {
-        $entity = self::emptyEntity();
-
-        $primaryPropertyName = $entity->entityInformation()->primaryPropertyName;
-        $primaryPropertyDataType = $entity->entityInformation()->informationForPropertyWithName($primaryPropertyName)->dataType;
-        if (!$primaryPropertyDataType->validate($primaryPropertyDataType->sanitizeValue($primaryPropertyValue), false)) throw new \obo\Exceptions\BadDataTypeException("Can't create entity from value " . (\is_scalar($primaryPropertyValue) ? "'" . print_r($primaryPropertyValue, true) . "'" : "") . " of '" . \gettype($primaryPropertyValue) . "' datatype. Primary property '" . $primaryPropertyName . "' in entity '" . self::classNameManagedEntity() . "' is of '" . $entity->entityInformation()->informationForPropertyWithName($primaryPropertyName)->dataType->name() . "' datatype.");
-        $entity->setValueForPropertyWithName($primaryPropertyValue, $primaryPropertyName);
-        $entity = \obo\obo::$identityMapper->mappedEntity($entity) ?: $entity;
+        $entity = self::mapedEntity($primaryPropertyValue);
```
Comment (lordjim666) [inline src/EntityManager.php]: mapped

File: src/EntityManager.php [hunk 3]
```diff
@@ -95,17 +110,15 @@ public static function emptyEntity() {
...
 
         if (!$entity->isInitialized()) {
             $data = self::rawDataForEntity($entity, $ignoreSoftDelete);
-            if (!count($data)) throw new \obo\Exceptions\EntityNotFoundException("Entity '" . self::classNameManagedEntity() . "' with primary property value '{$primaryPropertyName} = {$primaryPropertyValue}' does not exist in the repository or is deleted");
+            if (!count($data)) {
```
Comment (lordjim666) [inline src/EntityManager.php]: bool?

File: src/EntityManager.php [hunk 4]
```diff
@@ -119,18 +132,14 @@ public static function entityWithPrimaryPropertyValue($primaryPropertyValue, $ig
...
     public static function entityFromArray($data, $loadOriginalData = false, $overwriteOriginalData = true) {
-        $entity = self::emptyEntity();
-
-        $primaryPropertyName = $entity->entityInformation()->primaryPropertyName;
+        $entityClassName = self::classNameManagedEntity();
+        $primaryPropertyName = $entityClassName::entityInformation()->primaryPropertyName;
 
         if (isset($data[$primaryPropertyName]) OR \array_key_exists($primaryPropertyName, $data)) {
-            $entity->setValueForPropertyWithName($data[$primaryPropertyName], $primaryPropertyName);
+            $entity = self::mapedEntity($data[$primaryPropertyName]);
```
Comment (lordjim666) [inline src/EntityManager.php]: mapped

File: src/EntityManager.php (hunk 1 above)
Comment (hAdamm) [reply src/EntityManager.php]: Fixed

File: src/EntityManager.php (hunk 2 above)
Comment (hAdamm) [reply src/EntityManager.php]: Fixed

File: src/EntityManager.php (hunk 3 above)
Comment (hAdamm) [reply src/EntityManager.php]: Fixed

File: src/EntityManager.php (hunk 4 above)
Comment (hAdamm) [reply src/EntityManager.php]: Fixed

**Pull request merged**